compiler/optimizations/optimizeOnClauses.cpp

```diff
@@ -147,22 +178,27 @@ isFastPrimitive(CallExpr *call, bool isLocal) {
     // If this test is true, a remote get is required.
     if (!(call->get(1)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF) &&
           call->get(1)->getValType()->symbol->hasFlag(FLAG_WIDE_CLASS))) {
-      DEBUG_PRINTF(" *** OK (PRIM_WIDE_GET_LOCALE, etc.): %s\n",
-                   call->primitive->name);
-      return true;
+      return FAST_AND_LOCAL;
     }
-    break;
+    return FAST_NOT_LOCAL;
+
+  case PRIM_ARRAY_SHIFT_BASE_POINTER:
```

Comment:
Since my changes caused classifyPrimitive to be run over more code, I needed to add a handful of primitives that weren't previously handled in classifyPrimitive. This was one of them. If encountered with the old version, it would have just caused a halt.

I discovered and addressed all primitives missing from the switch statement by removing its default case (so that the end of the function causes the INT_ASSERT  if we ever got there). That way, the C compiler warns about missing cases. The cases that I added were ones I discovered in this way.

See my other comment about the local block I added to the module code for this case. I'm not sure I put the right body here. I'm interested in your opinion on what to do with it.